debezium-connector-postgres/src/main/java/io/debezium/connector/postgresql/PostgresConnectorTask.java

```diff
@@ -71,9 +75,13 @@ public class PostgresConnectorTask extends BaseSourceTask<PostgresPartition, Pos
     private volatile ErrorHandler errorHandler;
     private volatile PostgresSchema schema;
 
+    private Partition.Provider<PostgresPartition> partitionProvider = null;
+    private OffsetContext.Loader<PostgresOffsetContext> offsetContextLoader = null;
+
+    private final ReentrantLock stateLock = new ReentrantLock();
```

Comment:
I think this will create another instance of the `stateLock` and it won't synchronize with blocks guarded by `BaseSourceTask#stateLock`

Reply:
@vjuranek yes and is fine. I will add `PostgresConnectorTask#commitRecord(SourceRecord)` into avoid confusion for `DebeziumEngine` flow. `PostgresConnectorTask#commit` method is not using the state set by `BaseSourceTask#commitRecord`. So stateLock is used to make sure only one commit() processing is happening at any time. Let me know if you have any questions.

Reply:
Done.

Reply:
My main concern was that the commit can happen e.g. during stopping the connector. Looking more into it, it really seems harmless, we e.g. if the connection is already closed, the connector doesn't fail but skips the commit. So I'd suggest to rename this `stateLock` to e.g. `commitLock` to make clear that this has nothing to do with `stateLock` from parent class and it's intentionally used only for serializing calls of `commit()` method.

Reply:
Done.